### PR TITLE
Kill process when it doesn't stop cleanly and gather information (2.x)

### PR DESCRIPTION
**Pull request by cchesser**
Patching back the changes from issue #5 to the 2.x version. Some other changes includes:

* Had to pin to Chef 12.x for the tests (was taking latest, which on 13.x was getting some failures)
* Included changes from #21 as this was needed to support the tests from before (as there are two services included with naming patterns to expose the older bug)
* As part of the changes originally for #21 from #33, I update the tests which were failing from before, so those were also updated to get the tests to pass from before

Kitchen test results are [here](https://gist.github.com/cchesser/6513ef3aae4e99d4b693873c78d2a623).

### CODE REVIEW

**Comment by bbaugher** [comment]
+1

**Comment by bbaugher** [comment]
Looks like foodcritic updated and has found some lint so feel free to ignore it doesn't seem related

**Comment by cchesser** [comment]
Thanks @bbaugher!

**Comment by cchesser** [comment]
@bbaugher, would we be good with cutting a release for 2.4.0 which just included these changes? I assume there weren't others planned to be included on the release (based on what is currently documented on the milestone).

**Comment by bbaugher** [comment]
Yes I think that makes sense. We can try and get that done shortly after this is merged, possibly later this week

File: README.md [hunk 1]
```diff
@@ -46,7 +46,8 @@ Parameters:
  * `log_rotate_options`: A hash of options to configure log rotate. These will be merged with and override the defaults provided (view provider for defaults)
  * `version`: The version of tomcat to install. This effectively builds the tomcat_url from a known URL (`repo1.maven.org`) with the given version (default=`8.0.21`)
  * `tomcat_url`: The URL to the tomcat binary used to install tomcat. This will override the url provided by `version`. NOTE: `version` needs to still be up to date in order to install tomcat properly
- * `shutdown_timeout`: The timeout used when trying to shutdown the tomcat service (default = `60`)
+ * `shutdown_timeout`: The timeout used when trying to shutdown the tomcat service (default = `60`). If the timeout is reached, diagnostics are captured about the instance
+ and the a force shutdown is applied.
```
Comment (aujody) [inline README.md]: nit: grammar ...

**Comment by aujody** [comment]
+1 (other than the failing check... )

**Comment by mkwhitacre** [comment]
+1

File: README.md (hunk 1 above)
Comment (cchesser) [reply README.md]: I can correct, and can also put a PR out for master on this (since this was patching back 3.x changes).

Comment (cchesser) [reply README.md]: Applied: 70d6c40

**Comment by bbaugher** [comment]
Looks good, merging

**Pull request merged**